Parent/Parent/ObserverAlerts/ObserverAlertListViewController.swift

```diff
@@ -161,14 +163,50 @@ extension ObserverAlertListViewController: UITableViewDataSource, UITableViewDel
 
     func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
         let id = alerts[indexPath.row].id
+        let dismissTitle = String(localized: "Dismiss", bundle: .parent)
         return UISwipeActionsConfiguration(actions: [
-            UIContextualAction(style: .destructive, title: String(localized: "Dismiss", bundle: .parent)) { (_, _, completed) in
-                DismissObserverAlert(alertID: id).fetch { (_, _, error) in
-                    completed(error == nil)
-                }
+            UIContextualAction(style: .destructive, title: dismissTitle) { [weak self] (_, _, completion) in
+                self?.dismissAlert(id: id, completion: completion)
             }
         ])
     }
+
+    private func markAlertAsRead(id: String) {
+        interactor.markAlertAsRead(id: id)
+            .ignoreFailure()
+            .sink { [weak self] in
+                guard let self, let index = alerts.firstIndex(where: { $0.id == id }) else { return }
+
+                // making sure `read` state is set before UI updates
+                alerts[index].workflowState = .read
+
+                tableView.reloadRows(at: [.init(row: index, section: 0)], with: .automatic)
+                // If in the future this will be used in a SplitView, make sure the row is reselected.
+
+                updateTabBarBadgeCount()
+            }
+            .store(in: &subscriptions)
+    }
+
+    private func dismissAlert(id: String, completion: @escaping (Bool) -> Void) {
+        interactor.dismissAlert(id: id)
+            .sink(
+                receiveCompletion: {
+                    switch $0 {
+                    case .finished: completion(true)
+                    case .failure: completion(false)
+                    }
+                },
+                receiveValue: { [weak self] in
+                    guard let self, let index = alerts.firstIndex(where: { $0.id == id }) else { return }
+
+                    alerts.remove(at: index)
+                    tableView.deleteRows(at: [.init(row: index, section: 0)], with: .automatic)
+                    updateTabBarBadgeCount()
+                }
+            )
+            .store(in: &subscriptions)
+    }
```

Comment:
For future reference I would add some comments on why we need the local state updates.